src/Chapter4.hs

```diff
@@ -472,10 +485,12 @@ Implement the Applicative instance for our 'Secret' data type from before.
 -}
 instance Applicative (Secret e) where
     pure :: a -> Secret e a
-    pure = error "pure Secret: Not implemented!"
+    pure = Reward
 
     (<*>) :: Secret e (a -> b) -> Secret e a -> Secret e b
-    (<*>) = error "(<*>) Secret: Not implemented!"
+    (<*>) (Trap e) _ = Trap e
+    (<*>) _ (Trap e) = Trap e
+    (<*>) (Reward f) (Reward x) = Reward $ f x
```

Comment:
You can notice that it looks very like `Functor` instance. You could reuse `fmap` here instead of these two cases 👍🏼